### ISSUE TITLE
[Feature request] The ability to configure the sizes available for 'cycling'

### ISSUE THREAD

**Issue by Phoshi** [dawsers/hyprscroller#34]
I've switched to this plugin for my window management because I think it's likely to be more usable than traditional tiling window management for DQHD displays like the Odyssey G9 family, but at that resolution, the choice of 1/3rd as the minimum window width is, imo, too big. Windows would easily be useful at half the size. I know that I can still use the usual hyprland controls for changing window dimensions, but cycling between fractions that add up nicely is a very pleasant way of resizing windows.

How hard would it be to add a configuration line to set the presets? If it's a relatively easy 'first issue' for somebody that's never touched a hyprland plugin before, would you accept a PR?

**Comment by dawsers** [comment]
I think it's a reasonable request I can add.

In the meantime, you probably know there is the `fitsize` dispatch. You can open a number of windows and then make them fit the width of the monitor. I added that option specifically for ultra-wide monitors.

**Comment by dawsers** [comment]
This was implemented in 978c1ce8f64781826a0c0e1d7dea7448da0cc017

There are two new options for the plugin:

`column_widths`: determines the set of column widths *hyprscroller* will cycle through when resizing the width of a column in *row* mode. It is a string of any number of values chosen among: *onesixth, onefourth, onethird, onehalf, twothirds, one*. The default value is: *onehalf twothirds onethird*.

`window_heights`: determines the set of window heights *hyprscroller* will cycle through when resizing the height of a window in column mode. It is a string of any number of values chosen among: *onesixth, onefourth, onethird, onehalf, twothirds, one*. The default value is: *one onethird onehalf twothirds*.

So the default values are the "old" ones. If someone doesn't change their configuration, things work as they did in the past.

If you want to change them for an ultra-wide monitor, you could do something like:

```
plugin {
    scroller {
        # ultra-wide monitor
        column_widths = onefourth onethird onehalf onesixth
    }
}
```
